syntax = "proto3";

package billManagerAPI;

option go_package = "github.com/azuki774/bill-manager/internal/grpcapi";
import "google/protobuf/empty.proto";

service ElectConsumeService {
    rpc ElectConsumePost (OnedayElectConsume) returns (google.protobuf.Empty);
    rpc ElectConsumeGet (DateStruct) returns (OnedayElectConsume);
}

message OnedayElectConsume {
    DateStruct date = 1;
    int64 daytime = 2;
    int64 nighttime = 3;
    int64 total = 4;
}

message DateStruct {
    int32 year = 1;
    int32 month = 2;
    int32 day = 3;
}
